server/src/main/java/org/elasticsearch/action/admin/indices/resolve/TransportResolveClusterAction.java

```diff
@@ -0,0 +1,323 @@
+import org.elasticsearch.tasks.CancellableTask;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.ConnectTransportException;
+import org.elasticsearch.transport.NoSeedNodeLeftException;
+import org.elasticsearch.transport.NoSuchRemoteClusterException;
+import org.elasticsearch.transport.RemoteClusterAware;
+import org.elasticsearch.transport.RemoteClusterService;
+import org.elasticsearch.transport.TransportService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executor;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.elasticsearch.action.search.TransportSearchHelper.checkCCSVersionCompatibility;
+
+public class TransportResolveClusterAction extends HandledTransportAction<ResolveClusterActionRequest, ResolveClusterActionResponse> {
+
+    private static final Logger logger = LogManager.getLogger(TransportResolveClusterAction.class);
+    private static final String TRANSPORT_VERSION_ERROR_MESSAGE = "ResolveClusterAction requires at least Transport Version";
+
+    public static final String NAME = "indices:admin/resolve/cluster";
+    public static final ActionType<ResolveClusterActionResponse> TYPE = new ActionType<>(NAME, ResolveClusterActionResponse::new);
+    private final ThreadPool threadPool;
+    private final Executor searchCoordinationExecutor;
+    private final ClusterService clusterService;
+    private final RemoteClusterService remoteClusterService;
+    private final IndexNameExpressionResolver indexNameExpressionResolver;
+    private final boolean ccsCheckCompatibility;
+
+    @Inject
+    public TransportResolveClusterAction(
+        TransportService transportService,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver
+    ) {
+        super(NAME, transportService, actionFilters, ResolveClusterActionRequest::new, EsExecutors.DIRECT_EXECUTOR_SERVICE);
+        this.threadPool = threadPool;
+        this.searchCoordinationExecutor = threadPool.executor(ThreadPool.Names.SEARCH_COORDINATION);
+        this.clusterService = clusterService;
+        this.remoteClusterService = transportService.getRemoteClusterService();
+        this.indexNameExpressionResolver = indexNameExpressionResolver;
+        this.ccsCheckCompatibility = SearchService.CCS_VERSION_CHECK_SETTING.get(clusterService.getSettings());
+    }
+
+    @Override
+    protected void doExecute(Task task, ResolveClusterActionRequest request, ActionListener<ResolveClusterActionResponse> listener) {
+        // workaround for https://github.com/elastic/elasticsearch/issues/97916 - TODO remove this when we can
+        searchCoordinationExecutor.execute(ActionRunnable.wrap(listener, l -> doExecuteForked(task, request, l)));
```

Comment:
See this (resolved) discussion thread: https://github.com/elastic/elasticsearch/pull/102726#discussion_r1417770494

Reply:
The model was copied after the `TransportFieldCapabilitiesAction`, since that conformed to the several structural change requests that came from that discussion thread and several others.

Reply:
Thanks , I read the thread. @DaveCTurner what was the rationale behind suggesting SEARCH_COORDINATOR for this action? I feel like this action has little to do with search, and search coordinator is used for things like field caps and can match. Shouldn't we rather use the same thread pool that resolve index uses? Or is there some substantial difference between the two APIs that I am not seeing?

Reply:
> Shouldn't we rather use the same thread pool that resolve index uses? 

My original formulation copied the [ResolveIndexAction usage](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/action/admin/indices/resolve/ResolveIndexAction.java#L499), which runs it on the transport thread, which is what Jake/David were pointing out in [the original comment](https://github.com/elastic/elasticsearch/pull/102726#discussion_r1417770494).

Reply:
I don't have a particularly strong opinion about what executor we use here as long as it's not a `transport_worker` thread, and a vague leaning away from `GENERIC` since we don't really want to run 100+ of these in parallel. It's a bug that `ResolveIndexAction` doesn't fork too, this definitely shouldn't be running on a `transport_worker` either. Think of it like this: if it's lightweight enough to run on a `transport_worker`, it sort of doesn't matter where else it runs; conversely if you think this is too heavy to run on `SEARCH_COORDINATION` then it's _definitely_ too heavy to run on `transport_worker`.

Reply:
I agree with the need of forking. My concern was not around how lightweight or heavweight the action is, but more around what search coordination was introduced for, what it runs now (can match and field cas, and whether this action fits there (maybe it does from the perspective of being called from Kibana for CCS reasons, that are relevant to search). Perhaps it does. In our team discussion MANAGEMENT came up too. I am ok with either of the two at this point.

Reply:
Yeah `MANAGEMENT` could work too, although be warned that this pool is small (max 5 threads) and sometimes does fairly heavy stuff (e.g. collecting stats) so might add up to maybe 10s of seconds of latency sometimes.